core/sail/fts/lucene/src/test/java/org/eclipse/rdf4j/sail/lucene/AbstractGenericLuceneTest.java

```diff
@@ -0,0 +1,914 @@
+		// add the results of for the first query
+		results.add(resultSet);
+
+		// create a new result set
+		resultSet = new ArrayList<Map<String, String>>();
+
+		// one possible result
+		result = new HashMap<String, String>();
+		result.put("Resource", SUBJECT_1.stringValue());
+		result.put("Score", null); // null means: ignore the value
+		result.put("Snippet", "<B>five</B>");
+		resultSet.add(result);
+
+		// add the results of for the first query
+		results.add(resultSet);
+
+		evaluate(queries, results);
+	}
+
+	@Test
+	public void testSnippetQueries()
+		throws MalformedQueryException, RepositoryException, QueryEvaluationException
+	{
+		// prepare the query
+		// search for the term "one", but only in predicate 1
+		StringBuilder buffer = new StringBuilder();
+		buffer.append("SELECT \n");
+		buffer.append("  Resource, Score \n");
+		buffer.append("FROM \n");
+		buffer.append("  {Resource} <" + MATCHES + "> {} ");
+		buffer.append("    <" + QUERY + "> {\"one\"}; ");
+		buffer.append("    <" + PROPERTY + "> {<" + PREDICATE_1 + ">}; ");
+		buffer.append("    <" + SCORE + "> {Score} ");
+		String q = buffer.toString();
+
+		// fire the query
+		TupleQuery query = connection.prepareTupleQuery(QueryLanguage.SERQL, q);
+		TupleQueryResult result = query.evaluate();
+
+		// check the results
+		BindingSet bindings = null;
+
+		// the first result is subject 1 and has a score
+		int results = 0;
+		Set<IRI> expectedSubject = new HashSet<IRI>();
+		expectedSubject.add(SUBJECT_1);
+		expectedSubject.add(SUBJECT_2);
+		while (result.hasNext()) {
+			results++;
+			bindings = result.next();
+
+			// the resource should be among the set of expected subjects, if so,
+			// remove it from the set
+			assertTrue(expectedSubject.remove(bindings.getValue("Resource")));
+
+			// there should be a score
+			assertNotNull(bindings.getValue("Score"));
+		}
+
+		// there should have been only 2 results
+		assertEquals(2, results);
+
+		result.close();
+	}
+
+	/**
+	 * Test if the snippets do not accidentially come from the "text" field while we actually expect them to
+	 * come from the predicate field.
+	 */
+	@Test
+	public void testSnippetLimitedToPredicate()
+		throws MalformedQueryException, RepositoryException, QueryEvaluationException
+	{
+		// more test-data
+		RepositoryConnection myconnection = repository.getConnection();
+		myconnection.begin();
+		// we use the string 'charly' as test-case. the snippets should contain
+		// "come" and "unicorn"
+		// and 'poor' should not be returned if we limit on predicate1
+		// and watch http://www.youtube.com/watch?v=Q5im0Ssyyus like 25mio others
+		myconnection.add(SUBJECT_1, PREDICATE_1, vf.createLiteral("come charly lets go to candy mountain"));
+		myconnection.add(SUBJECT_1, PREDICATE_1, vf.createLiteral("but the unicorn charly said to goaway"));
+		myconnection.add(SUBJECT_1, PREDICATE_2, vf.createLiteral("there was poor charly without a kidney"));
+		myconnection.commit();
+		myconnection.close();
+
+		{
```

Comment:
RDF4J does not use brace blocks like this. If the code is difficult to understand otherwise you need to split it up into smaller sub-methods